swayfmt/src/utils/language/attribute.rs

```diff
@@ -40,41 +40,43 @@ impl Format for AttributeDecl {
-            if let Some(Some(doc_comment)) = attr
-                .args
-                .as_ref()
-                .map(|args| args.inner.final_value_opt.as_ref())
-            {
-                writeln!(formatted_code, "///{}", doc_comment.as_str().trim_end())?;
+        // `#`
+        write!(formatted_code, "{}", self.hash_token.span().as_str())?;
+        // `[`
+        Self::open_square_bracket(formatted_code, formatter)?;
+        for attr in self.attribute.get().into_iter() {
+            if attr.name.as_str() == "doc" {
+                if let Some(Some(doc_comment)) = attr
+                    .args
+                    .as_ref()
+                    .map(|args| args.inner.final_value_opt.as_ref())
+                {
+                    writeln!(formatted_code, "///{}", doc_comment.as_str().trim_end())?;
+                }
+            } else {
+                formatter.with_shape(
+                    formatter.shape.with_default_code_line(),
+                    |formatter| -> Result<(), FormatterError> {
+                        // name e.g. `storage`
+                        write!(formatted_code, "{}", attr.name.span().as_str())?;
+                        if let Some(args) = &attr.args {
+                            // `(`
+                            Self::open_parenthesis(formatted_code, formatter)?;
+                            // format and add args e.g. `read, write`
+                            args.get().format(formatted_code, formatter)?;
+                            // ')'
+                            Self::close_parenthesis(formatted_code, formatter)?;
+                        };
+                        write!(formatted_code, "{} ", PunctKind::Comma.as_char())?;
+                        Ok(())
+                    },
+                )?;
             }
-        } else {
-            formatter.with_shape(
-                formatter.shape.with_default_code_line(),
-                |formatter| -> Result<(), FormatterError> {
-                    // `#`
-                    write!(formatted_code, "{}", self.hash_token.span().as_str())?;
-                    // `[`
-                    Self::open_square_bracket(formatted_code, formatter)?;
-                    // name e.g. `storage`
-                    write!(formatted_code, "{}", attr.name.span().as_str())?;
-                    if let Some(args) = &attr.args {
-                        // `(`
-                        Self::open_parenthesis(formatted_code, formatter)?;
-                        // format and add args e.g. `read, write`
-                        args.get().format(formatted_code, formatter)?;
-                        // ')'
-                        Self::close_parenthesis(formatted_code, formatter)?;
-                    };
-                    // `]\n`
-                    Self::close_square_bracket(formatted_code, formatter)?;
-
-                    Ok(())
-                },
-            )?;
         }
-
+        formatted_code.pop();
```

Comment:
If you can find a way around using `.pop()` or destructive means it will narrow down instances of mysterious formatter failures creating unparsable code! There may be other instances that are in the formatter at present, but we should try to avoid adding any more and look to review the others at some point.